src/engine/cachingreader.cpp

```diff
@@ -222,7 +222,16 @@ void CachingReader::process() {
     }
 }
 
-SINT CachingReader::read(SINT sample, bool reverse, SINT numSamples, CSAMPLE* buffer) {
+SINT CachingReader::read(SINT start_sample, bool reverse, SINT numSamples, CSAMPLE* buffer) {
```

Comment:
The parameters start_sample and numSample belong closer together. The parameter reverse controls how this chunk of samples is copied into the buffer. Order: startSample, numSamples, reverse, buffer.

I'm still unsure which sample should be denoted "start" or "first" in case of reverse directions?!? When switching from sample to frame positions I would recommend to add a new class that exactly defines how a consecutive number of (complete) sample frames is specified, i.e. which frame is denoted as the first/start frame. One member of this class would be the direction, e.g. enum class <SomeGenericPrefix: "Audio"?>Direction { FORWARD, REVERSE }. Even in the current code base all those reverse flags could be replaced step by step by such a type-safe enum class.